lib/jekyll-gist/gist_tag.rb

```diff
@@ -4,7 +4,12 @@ class GistTag < Liquid::Tag
 
       def render(context)
         if tag_contents = determine_arguments(@markup.strip)
-          gist_id, filename = tag_contents[0], tag_contents[1]
+          gist_id = context.registers[:page]['gist_id']
```

Comment:
> the gist line doesn't interpret any variables

That's what I am suggesting: try to interpret the parameters as variables and if that doesn't return anything useful then just use them as is.